packages/connect/src/connect.js

```diff
@@ -1,140 +1,193 @@
 
-const mapStateToStores = (state) => {
-  // find store properties and map them to their none observable raw value
-  // to do not trigger none static this.setState calls
-  // from the static getDerivedStateFromProps lifecycle method
-  const component = state[COMPONENT];
-  return Object.keys(component)
-    .map((key) => component[key])
-    .filter(isObservable)
-    .map(raw);
+/**
+ * The default options of connect.
+ */
+const defaultOptions = {
+  /**
+   * Define if `connect` should inject the store props in the component, or only
+   * make it reactive. Useful when the store is accessed via `useConnect` and
+   * the props are passed down an HTML node.
+   */
+  injectProps: true,
 };
 
+/**
+ * Connect a React component with the Frontity connect store and make it
+ * reactive to its changes.
+ *
+ * @param Comp - The React component to be connected.
+ * @param options - The connect options, defined in {@link defaultOptions}.
+ *
+ * @returns The same Component, but reactive and with the store included in the
+ * props.
+ */
 export function connect(Comp, options) {
-  const isStatelessComp = !(Comp.prototype && Comp.prototype.isReactComponent);
-
-  let ReactiveComp;
-
-  const defaultOptions = {
-    injectProps: true,
-  };
-
+  const isFunctionalComp = !(Comp.prototype && Comp.prototype.isReactComponent);
   options = options ? { ...defaultOptions, ...options } : defaultOptions;
 
-  if (isStatelessComp && hasHooks) {
-    // use a hook based reactive wrapper when we can
-    ReactiveComp = memo((props) => {
-      // set a flag to know when the component is unmounted.
-      const _isMounted = useRef(true);
+  let ReactiveComp;
 
-      // use a dummy setState to update the component
+  if (isFunctionalComp) {
+    // Use a hook based reactive wrapper.
+    ReactiveComp = (props) => {
+      // Dummy setState to update the component.
       const [, setState] = useState();
 
-      // get frontity from the context;
-      const frontity = useContext(context);
+      // Use a ref to store the reaction because we want to do some annotations.
+      const reaction = useRef();
 
-      // create a memoized reactive wrapper of the original component (render)
-      // at the very first run of the component function
+      // Create a memoized reactive wrapper of the original component (render)
+      // at the very first run of the component function.
       const render = useMemo(
-        () =>
-          observe(Comp, {
-            scheduler: () => _isMounted.current && setState({}),
+        () => {
+          reaction.current = observe(Comp, {
+            scheduler: () => {
+              // Trigger a new rerender if the component has already been
+              // mounted.
+              if (reaction.current.mounted) setState({});
+              // Annotate it as "changed" if the component has not been mounted
+              // yet.
+              else reaction.current.changedBeforeMounted = true;
+            },
```

Comment:
Note to self: The custom scheduler is used here in order to fix the bug: https://github.com/RisingStack/react-easy-state/issues/196#issuecomment-766367636